test/linalg/special.jl

```diff
@@ -103,6 +103,17 @@ let a=[1.0:n;]
        @test_approx_eq full(B - convert(Spectype,A)) full(B - A)
        @test_approx_eq full(convert(Spectype,A) - B) full(A - B)
    end
+
+   C = rand(n,n)
+   for TriType in [Base.LinAlg.UnitLowerTriangular, Base.LinAlg.UnitUpperTriangular, UpperTriangular, LowerTriangular]
+       D = TriType(C)
+       for Spectype in [Diagonal, Bidiagonal, Tridiagonal, Matrix]
+           @test full(D + convert(Spectype,A)) ≈ full(D + A)
```

Comment:
Consistency with the above tests. I can try them out with exact and report back (they should pass, I agree).